internal/api/cluster.go

```diff
@@ -108,29 +107,23 @@ func handleCreateCluster(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if createClusterRequest.EKSConfig != nil && c.Provisioner.ProvisionerType() != provisioner.EKSProvisionerType {
-		c.Logger.Error("invalid request: specified EKSConfig when provisioner type is not 'eks")
-		w.WriteHeader(http.StatusBadRequest)
-		return
-	}
-	if c.Provisioner.ProvisionerType() == provisioner.EKSProvisionerType && createClusterRequest.EKSConfig == nil {
-		c.Logger.Error("invalid request: EKSConfig is required when provisioner type is 'eks")
-		w.WriteHeader(http.StatusBadRequest)
-		return
+	provisioner := "kops"
+	if createClusterRequest.Provisioner != "" {
+		provisioner = createClusterRequest.Provisioner
```

Comment:
Maybe I missed it, but is this validated anywhere so that I can't send a random value? We should maybe do that and also maybe used constant values for the strings?

Reply:
Thank you for your review! I appreciate your suggestion to include validation, which I intentionally deferred to the next PR. However, I want to assure you that the current usage is limited to `eks` or `kops` (default), so sending a random value should not cause any issue. Thank you again for taking the time to review!